Projects/GeometryTestProject/ExcludePropertiesClass.cs

```diff
@@ -11,56 +11,43 @@ namespace GeometryTestProject
   public class ExcludePropertiesClass
   {
     [TestMethod]
-    public void TestExcludeName_Find()
+    public void TestExcludeTwoFields()
     {
       DataQueryBuilder queryBuilder = DataQueryBuilder.Create();
-      queryBuilder.AddAllProperties();
-      queryBuilder.AddProperty( "trim( name )" );
-      queryBuilder.ExcludeProperty( "name" );
+      queryBuilder.AddProperties( "*" );
+      queryBuilder.ExcludeProperties( "name", "age" );
 
       IList<Dictionary<string, object>> res = Backendless.Data.Of( "A" ).Find( queryBuilder );
 
-      foreach ( Dictionary<string, object> kvp in res )
-        if ( !kvp.ContainsKey( "name" ) )
-          Assert.IsTrue( true );
-        else
-          Assert.IsTrue( false );
+      Assert.IsTrue( !res[ 0 ].ContainsKey( "name" ), "First object is contains key 'name'" );
+      Assert.IsTrue( !res[ 0 ].ContainsKey( "age" ), "First object is contains key 'age'" );
+
+      Assert.IsTrue( !res[ 1 ].ContainsKey( "name" ), "First object is contains key 'name'" );
+      Assert.IsTrue( !res[ 1 ].ContainsKey( "age" ), "First object is contains key 'age'" );
     }
 
     [TestMethod]
-    public void TestExcludeName_Find_WithStar()
+    public void TestCreateFieldTIME()
     {
       DataQueryBuilder queryBuilder = DataQueryBuilder.Create();
-      queryBuilder.AddProperties( "*" );
-      queryBuilder.AddProperty( "trim( name )" );
-      queryBuilder.ExcludeProperty( "name" );
+      queryBuilder.AddProperties( "*", "TIME(created)" );
 
       IList<Dictionary<string, object>> res = Backendless.Data.Of( "A" ).Find( queryBuilder );
-      foreach ( Dictionary<string, object> kvp in res )
-        if ( !kvp.ContainsKey( "name" ) )
-          Assert.IsTrue( true );
-        else
-          Assert.IsTrue( false );
+
+      Assert.IsTrue( res[ 0 ].ContainsKey( "Time" ), "First object does not contain 'Time' key" );
+      Assert.IsTrue( res[ 1 ].ContainsKey( "Time" ), "Second object does not contain 'Time' key" );
     }
 
     [TestMethod]
-    public void TestExcludeNameAndLocation_Find_Related()
+    public void TestRelatedField()
     {
       DataQueryBuilder queryBuilder = DataQueryBuilder.Create();
-      queryBuilder.AddProperties( "*", "table_B.adress", "TIME(created)" );
-      queryBuilder.ExcludeProperties( "name", "location" );
+      queryBuilder.AddProperties( "*", "table_B.adress" );
 
       IList<Dictionary<string, object>> res = Backendless.Data.Of( "A" ).Find( queryBuilder );
 
-      foreach ( Dictionary<string, object> kvp in res )
-        if ( ( String )kvp["adress"] == "Tom Street" || ( String ) kvp["adress"] == "Curse Street" )
-        {
-          if ( kvp.ContainsKey( "Time" ) )
-            if ( !kvp.ContainsKey( "name" ) )
-              Assert.IsTrue( true );
-        }
-        else
-          Assert.IsTrue( false );
+      Assert.IsTrue( (String) res[ 0 ][ "adress" ] == "Tom Street", "First object does not contain 'adress' field" );
```

Comment:
I do not like the test for "Tom Street" because it is not clear why there would be that value. Instead check for the presence of that property.